packages/backend/src/apps/tiles/actions/create-row/index.ts

```diff
@@ -78,7 +78,17 @@ const action: IRawAction = {
       tableId: string
       rowData: { columnId: string; cellValue: string }[]
     }
-    await validateTileAccess($.flow?.userId, tableId as string)
+
+    try {
+      await TableCollaborator.hasAccess($.flow?.userId, tableId, 'editor')
+    } catch (e) {
+      throw new StepError(
+        'You do not have access to this tile',
+        'Ensure you have access to the tile you are trying to update a row in.',
+        $.step.position,
+        $.app.name,
+      )
+    }
```

Comment:
small nit: would be nice to refactor this as a constant or helper function since it is used across multiple actions